Client/Ecosia/UI/NTP/Impact/NTPImpactRowView.swift

```diff
@@ -0,0 +1,172 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+final class NTPImpactRowView: UIView, NotificationThemeable {
+    struct UX {
+        static let height: CGFloat = 80
+        static let cornerRadius: CGFloat = 10
+        static let horizontalSpacing: CGFloat = 8
+        static let padding: CGFloat = 16
+        static let imageHeight: CGFloat = 48
+        static let imageHeightWithProgress: CGFloat = 26
+        static let progressWidth: CGFloat = 48
+        static let progressHeight: CGFloat = 30
+        static let progressLineWidth: CGFloat = 2
+    }
+    
+    private lazy var imageContainer: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    private lazy var imageView: UIImageView = {
+        let image = UIImageView()
+        image.translatesAutoresizingMaskIntoConstraints = false
+        image.contentMode = .scaleAspectFit
+        return image
+    }()
+    private lazy var totalProgressView: ProgressView = {
+        ProgressView(size: .init(width: UX.progressWidth, height: UX.progressHeight),
+                     lineWidth: UX.progressLineWidth)
+    }()
+    private lazy var currentProgressView: ProgressView = {
+        ProgressView(size: .init(width: UX.progressWidth, height: UX.progressHeight),
+                     lineWidth: UX.progressLineWidth)
+    }()
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel()
+        label.font = .preferredFont(forTextStyle: .title2).bold()
```

Comment:
I think you need to add `adjustsFontSizeToFitWidth` to make the label scale - same for subtitleLabel and actionButton